
This reverts the parts of:

commit 86fe2db84ba7b9a83b60323cd805198ff43136ec
Author: Bastien Nocera <hadess@hadess.net>
Date:   Fri Mar 20 18:28:08 2015 +0100

    cfg80211: Update for newest mac80211 API


that breaks building with 3.19 series kernels.



Signed-off-by: Thomas Backlund <tmb@mageia.org>


diff -Nurp linux/3rdparty/rtl8723bs.rev/include/ioctl_cfg80211.h linux/3rdparty/rtl8723bs/include/ioctl_cfg80211.h
--- linux/3rdparty/rtl8723bs.rev/include/ioctl_cfg80211.h	2015-05-07 17:46:55.000000000 +0300
+++ linux/3rdparty/rtl8723bs/include/ioctl_cfg80211.h	2015-05-10 14:55:44.744038033 +0300
@@ -22,10 +22,6 @@
 
 #include <linux/version.h>
 
-#if (LINUX_VERSION_CODE < KERNEL_VERSION(4, 0, 0))
-#error This driver only works with kernel 4.0
-#endif
-
 struct rtw_wdev_invit_info {
 	u8 state; /* 0: req, 1:rep */
 	u8 peer_mac[ETH_ALEN];
diff -Nurp linux/3rdparty/rtl8723bs.rev/os_dep/ioctl_cfg80211.c linux/3rdparty/rtl8723bs/os_dep/ioctl_cfg80211.c
--- linux/3rdparty/rtl8723bs.rev/os_dep/ioctl_cfg80211.c	2015-05-10 14:54:53.241616597 +0300
+++ linux/3rdparty/rtl8723bs/os_dep/ioctl_cfg80211.c	2015-05-10 14:49:42.766354711 +0300
@@ -1289,16 +1289,16 @@ static int cfg80211_rtw_get_station(stru
 			goto exit;
 		}
 
-		sinfo->filled |= BIT(NL80211_STA_INFO_SIGNAL);
+		sinfo->filled |= STATION_INFO_SIGNAL;
 		sinfo->signal = translate_percentage_to_dbm(padapter->recvpriv.signal_strength);
 
-		sinfo->filled |= BIT(NL80211_STA_INFO_TX_BITRATE);
+		sinfo->filled |= STATION_INFO_TX_BITRATE;
 		sinfo->txrate.legacy = rtw_get_cur_max_rate(padapter);
 
-		sinfo->filled |= BIT(NL80211_STA_INFO_RX_PACKETS);
+		sinfo->filled |= STATION_INFO_RX_PACKETS;
 		sinfo->rx_packets = sta_rx_data_pkts(psta);
 
-		sinfo->filled |= BIT(NL80211_STA_INFO_TX_PACKETS);
+		sinfo->filled |= STATION_INFO_TX_PACKETS;
 		sinfo->tx_packets = psta->sta_stats.tx_pkts;
 
 	}
@@ -2473,6 +2473,7 @@ void rtw_cfg80211_indicate_sta_assoc(_ad
 			ie_offset = _REASOCREQ_IE_OFFSET_;
 	
 		sinfo.filled = 0;
+		sinfo.filled = STATION_INFO_ASSOC_REQ_IES;
 		sinfo.assoc_req_ies = pmgmt_frame + WLAN_HDR_A3_LEN + ie_offset;
 		sinfo.assoc_req_ies_len = frame_len - WLAN_HDR_A3_LEN - ie_offset;
 		cfg80211_new_sta(ndev, GetAddr2Ptr(pmgmt_frame), &sinfo, GFP_ATOMIC);
@@ -3078,7 +3079,8 @@ static int	cfg80211_rtw_dump_station(str
 		goto exit;
 	}
 	memcpy(mac, psta->hwaddr, ETH_ALEN);
-	sinfo->filled = BIT(NL80211_STA_INFO_SIGNAL);
+	sinfo->filled = 0;
+	sinfo->filled |= STATION_INFO_SIGNAL;
 	sinfo->signal = psta->rssi;
 	
 exit:
